src/Symfony/Component/Security/Core/User/AttributesBasedUserProviderInterface.php

```diff
@@ -18,6 +18,9 @@
  * This is particularly useful with self-contained access tokens.
  *
  * @experimental
+ *
+ * @template TUser of UserInterface
```

Comment:
this should be `@template-covariant` to be more flexible as the template type is only used in covariant positions